File: src/main/java/com/tomath/questao/QuestaoService.java
package com.tomath.questao;

import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class QuestaoService {
    final QuestaoRepository questaoRepository;

    public QuestaoService(QuestaoRepository questaoRepository) {
        this.questaoRepository = questaoRepository;
    }

    public List<Questao> listarQuestoes() {
        return questaoRepository.findAll();
    }

    public Long salvarQuestao(Questao questao) {
        return questaoRepository.save(questao).getId();
    }
}

File: src/main/kotlin/com/tomath/questao/QuestaoService.kt
package com.tomath.questao

import org.springframework.stereotype.Service

@Service
class QuestaoService(val questaoRepository: QuestaoRepository) {
    fun listarQuestoes(): List<Questao> {
        return questaoRepository.findAll() as List<Questao>
    }

    fun salvarQuestao(questao: Questao): Long {
        return questaoRepository.save(questao).id!!
    }
}